Client/WebFilters/CosmeticFiltersResourceDownloader.swift

```diff
@@ -0,0 +1,271 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Shared
+import BraveShared
+
+private let log = Logger.browserLogger
+
+private struct CosmeticFilterNetworkResource {
+    let resource: CachedNetworkResource
+    let fileType: FileType
+    let type: CosmeticFiltersResourceDownloader.CosmeticFilterType
+}
+
+class CosmeticFiltersResourceDownloader {
+    static let shared = CosmeticFiltersResourceDownloader()
+    
+    private let networkManager: NetworkManager
+    
+    static let folderName = "cmf-data"
+    private let servicesKeyName = "SERVICES_KEY"
+    private let servicesKeyHeaderValue = "BraveServiceKey"
+    private var engine = AdblockRustEngine()
+    
+    static let endpoint = "https://raw.githubusercontent.com/iccub/brave-ios/development"
```

Comment:
I assume this will be replaced?

Reply:
Yes, we do not have a proper endpoints set up yet, will try to push for it today.